### PR TITLE
Test against Node 20, update tests for compatibility

### PR DESCRIPTION
**Pull request by trevor-scheer**
Test against Node 20 and update tests as needed.

This unblocks our integrations from testing against Node 20 as well since the testsuite is part of this changeset / has Node 20 test failures.

### CODE REVIEW

**Comment by trevor-scheer** [comment]
Stuck on the last couple tests where the connection shouldn't be closing but it is (for Node 20). [This PR](https://github.com/nodejs/node/pull/46333) seems to be related, but my attempts to apply the information provided haven't been fruitful so far. I've tried adding the `transfer-encoding: 'chunked'` header in the response.

File: packages/integration-testsuite/src/httpServerTests.ts [hunk 1]
```diff
@@ -382,7 +382,7 @@ export function defineIntegrationTestSuiteHttpServerTests(
         return req.then((res) => {
           expect(res.status).toEqual(400);
           expect(
-            ['Unexpected token f', 'Bad Request', 'Invalid JSON'].some(
+            ['in JSON at position 1', 'Bad Request', 'Invalid JSON'].some(
```
Comment (trevor-scheer) [inline packages/integration-testsuite/src/httpServerTests.ts:385]: Error message changed in Node 20, this substring is compatible across all of our supported versions.

File: packages/server/src/__tests__/express4/expressSpecific.test.ts [hunk 2]
```diff
@@ -193,7 +193,7 @@ it('supporting doubly-encoded variables example from migration guide', async ()
       query: 'query Hello($s: String!){hello(s: $s)}',
       variables: '{malformed JSON}',
     })
-    .expect(400, 'Unexpected token m in JSON at position 1');
+    .expect(400, /in JSON at position 1/);
```
Comment (trevor-scheer) [inline packages/server/src/__tests__/express4/expressSpecific.test.ts:196]: Error message changed in Node 20, this substring is compatible across all of our supported versions.

File: packages/server/src/__tests__/plugin/drainHttpServer/stoppable.test.ts [hunk 3]
```diff
@@ -113,7 +112,7 @@ Object.keys(schemes).forEach((schemeName) => {
         const err = await a.failure(
           request(`${schemeName}://localhost:${p}`).agent(scheme.agent()),
         );
-        expect(err.message).toMatch(/ECONNREFUSED/);
+        expect(err.code).toMatch(/ECONNREFUSED/);
```
Comment (trevor-scheer) [inline packages/server/src/__tests__/plugin/drainHttpServer/stoppable.test.ts:115]: `.message` is undefined in Node 20 but `.code` is defined across all of our supported versions.

File: packages/server/src/__tests__/plugin/drainHttpServer/stoppable.test.ts [hunk 4]
```diff
@@ -135,8 +134,10 @@ Object.keys(schemes).forEach((schemeName) => {
             scheme.agent({ keepAlive: true }),
           ),
         );
-        expect(err.message).toMatch(/ECONNREFUSED/);
-        expect(closed).toBe(0);
+        expect(err.code).toMatch(/ECONNREFUSED/);
+
+        const isNode20 = !!process.version.match(/^v20\./);
+        expect(closed).toBe(isNode20 ? 1 : 0);
```
Comment (trevor-scheer) [inline packages/server/src/__tests__/plugin/drainHttpServer/stoppable.test.ts:140]: (currently) unexplained change in behavior in `http(s).Server`, but maybe this isn't actually important for us to validate/enforce. https://github.com/nodejs/node/pull/46333 seems to be the only relevant change listed in the Node 20 changelog but I haven't been able to apply the information there in a useful way that "corrects" this test. That PR also only seems to apply to when expected headers don't exist, which isn't the case here.

File: packages/server/src/__tests__/plugin/drainHttpServer/stoppable.test.ts [hunk 5]
```diff
@@ -301,12 +302,21 @@ Object.keys(schemes).forEach((schemeName) => {
 
     if (schemeName === 'http') {
       it('with in-flights finishing before grace period ends', async () => {
```
Comment (trevor-scheer) [inline packages/server/src/__tests__/plugin/drainHttpServer/stoppable.test.ts:304]: This test was causing an open handle / `AggregateError` that I was unfruitful in tracking down. Seems to have been related to spawning a process (maybe the server in the process wasn't being killed?). In any case, I think this change simplifies the test a bit by eliminating the need to spawn a process.

File: packages/server/src/__tests__/plugin/drainHttpServer/stoppable.test.ts (hunk 4 above)
Comment (glasser) [reply packages/server/src/__tests__/plugin/drainHttpServer/stoppable.test.ts:140]: Or https://github.com/nodejs/node/pull/46331 ?

Comment (trevor-scheer) [reply packages/server/src/__tests__/plugin/drainHttpServer/stoppable.test.ts:145]: For future reference, @glasser helped me pin this to changes made in v19 and v20.3
(http) https://github.com/nodejs/node/pull/43522
(https equiv) https://github.com/nodejs/node/pull/48383

**Pull request merged**